src/Lean/Parser/Command.lean

```diff
@@ -147,7 +147,7 @@ def openDecl         := withAntiquot (mkAntiquot "openDecl" `Lean.Parser.Command
   openHiding <|> openRenaming <|> openOnly <|> openSimple <|> openScoped
 @[builtinCommandParser] def «open»    := leading_parser withPosition ("open " >> openDecl)
 
-@[builtinCommandParser] def «mutual» := leading_parser "mutual " >> many1 (ppLine >> notSymbol "end" >> commandParser) >> ppDedent (ppLine >> "end") >> terminationSuffix
+@[builtinCommandParser] def «mutual» := leading_parser "mutual " >> many1 (ppLine >> notSymbol "end" >> withSyntaxOfStack commandParser) >> ppDedent (ppLine >> "end") >> terminationSuffix
```

Comment:
Repeatedly elaborating the notations in each iteration could be solved by a fusion of `withSyntaxOfStack` and `many1`, but it is hard to imagine this being relevant in practice.